garden_ai/model_connectors/model_utils.py

```diff
@@ -0,0 +1,117 @@
+from typing import Dict, Type, Union, Optional
+
+from pydantic import (
+    HttpUrl,
+)
+
+from .exceptions import (
+    ConnectorInvalidUrlError,
+    UnsupportedConnectorError,
+)
+from .github_conn import GitHubConnector
+from .hugging_face import HFConnector
+from .model_connector import ModelConnector, ModelRepository
+
+
+# Maps ModelRepositorys to their corresponding connector class.
+CONNECTOR_MAPPING: Dict[ModelRepository, Type[ModelConnector]] = {
+    ModelRepository.GITHUB: GitHubConnector,
+    ModelRepository.HUGGING_FACE: HFConnector,
+}
+
+
+def match_connector_type_by_url(url: str) -> Optional[Type[ModelConnector]]:
+    """Match url to the appropriate ModelRepository
+
+    Args:
+        url: A full HTTP URL as a str
+
+    Returns:
+        ModelConnector: The appropriate ModelConnector type for the URL.
+
+    Raises:
+        ValueError: When the URL is not able to be matched with a supported repository.
+    """
+    url = str(ModelConnector.is_valid_url(url))
+    if url:
+        if "github.com" in url:
+            return CONNECTOR_MAPPING.get(ModelRepository.GITHUB)
+        elif "huggingface.co" in url:
+            return CONNECTOR_MAPPING.get(ModelRepository.HUGGING_FACE)
+        else:
+            raise ValueError("Repository type is not supported.")
+    else:
+        raise ValueError("Invalid URL")
+
+
+def match_connector_type_by_id_and_type(
+    repo_id: str,
+    repo_type: str,
+) -> Optional[Type[ModelConnector]]:
+    """Match repo_id to the appropriate ModelRepository
+
+    Args:
+        repo_id (str): A repository id in the form 'owner/repo'
+        repo_type (str): A repository type, e.g 'GH' for GitHub, 'HF' for Hugging Face
+
+    Raises:
+        ValueError: When the the given repo_type is unsupported.
+    """
+    if ModelConnector.validate_repo_id(repo_id):
+        if repo_type == "GH":
+            return CONNECTOR_MAPPING.get(ModelRepository.GITHUB)
+        if repo_type == "HF":
+            return CONNECTOR_MAPPING.get(ModelRepository.HUGGING_FACE)
+        else:
+            raise ValueError("Unsupported repo_type")
+    else:
+        raise ValueError("Invalid repo_id.")
+
+
+def create_connector(
```

Comment:
I've mentioned it already but I'm a big fan of this UX. I feel like the `create_connector(my_url)` usage is going to be how 99% of model connectors are created now

Reply:
another hot take: I like the url-only instantiation so much I almost think we could unburden this by not supporting `repo_type='HF'` mode of calling this. Do we need to support `create_connector("Garden-AI/sklearn-iris", repo_type='HF')` when users that want more control and/or don't want to instantiate from a URL can directly instantiate with `HuggingFaceConnector("Garden-AI/sklearn-iris")`?